core/src/replay_stage.rs

```diff
@@ -2400,16 +2401,23 @@ impl ReplayStage {
         let new_root = tower.record_bank_vote(bank);
 
         if let Some(new_root) = new_root {
+            let highest_super_majority_root = Some(
+                block_commitment_cache
+                    .read()
+                    .unwrap()
+                    .highest_super_majority_root(),
+            );
             Self::check_and_handle_new_root(
```

Comment:
Out of paranoia, I inlined all of this and plugged into a diff checker. Code looks identical except for the following:

1. We pass in `parent_slot` instead of `vote_bank` <-- seems like a good change
2. We pass in `highest_super_majority_root` instead of `block_commitment_cache` <-- again, seems good
3. Reverting to positional formatting of arguments for `err` <-- I think we should go back to string capture style
4. Printing out `pubkey` when setting root <-- we pass `pubkey` through a couple API layers just for this.. is it worth it?

Reply:
agreed, would be nice if we could have some log magic to tag on pubkey in local cluster haha